lib/cli/input/eval.js

```diff
@@ -0,0 +1,71 @@
+const inquirer = require('inquirer');
+const PromptGenerator = require('../../utils/PromptGenerator');
+const preferenceManager = require('../../utils/preference-manager');
+const path = require('path');
+
+const defaultPrefPath = path.join(__dirname, '../../../default-prefs.json');
+const defaultPrefs = JSON.parse(require('fs').readFileSync(defaultPrefPath, 'utf8'));
+const {supportedLanguages} = defaultPrefs;
+
+const getLengthValidator = (invalidMessage) => {
+  return (value) => {
+    if (value.length) {
+      return true;
+    }
+    else {
+      return invalidMessage;
+    }
+  }
+}
+
+const getLabPrompt = async () => {
```

Comment:
The name of the variable seems a bit off. Since this function helps receive all the lab info (lab name, commit hash, preferred language), it may be better to call this function, `getSubmitInfo`.